src/datachain/toolkit/split.py

```diff
@@ -1,7 +1,16 @@
-from datachain import C, DataChain
+from typing import Optional
 
+from datachain import DataChain
+from datachain.func import bit_and, bit_xor, int_hash_64
```

Comment:
Is it an universal function among all DBs?

Ideally, dc.split(0.2, seed=20241203) should return the same result for the same dataset in locally as well as in cloud.

Reply:
> Is it an universal function among all DBs?

`int_hash_64` implementation is specific for ClickHouse. I was able to implement it for SQLite, but I am not 100% sure about other databases :(

Other problem is with int64 in SQLite, it only supports signed int64, this requires additional steps and attention.

One of the possible solution for all databases will be using md5/halfmd5/sha256, but these are slow.

> Ideally, dc.split(0.2, seed=20241203) should return the same result for the same dataset in locally as well as in cloud.

That was my goal, and it works now for both SQLite and ClickHouse.

Reply:
Resolved with #678